src/KafkaFlow/ActivityFactory.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.Diagnostics;
+
+namespace KafkaFlow
+{
+    internal class ActivityFactory : IActivityFactory
+    {
+        public Activity Start(string topicName, ActivityOperationType activityOperationType, ActivityKind activityKind)
```

Comment:
Can't this be static?

Reply:
It can as it ended up having no state. Pls review https://github.com/Farfetch/kafkaflow/pull/467/commits/9b9bb1b2bc2523ce987dd95e3fc0bf53e0852748